In [1]:
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score

In [2]:
# Load the dataset
df = pd.read_csv('/Users/sofiatajide/Downloads/Bike-Sharing-Dataset/day.csv')

In [3]:
# Define the independent variables
X = df[['season', 'temp', 'hum', 'windspeed', 'weathersit']]

# Define the dependent variable
y = df['cnt']

In [4]:
# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [5]:
# Fit the linear regression model on the training data
model = LinearRegression()
model.fit(X_train, y_train)

LinearRegression()

In [6]:
# Make predictions on the testing data
y_pred = model.predict(X_test)

In [7]:
# Evaluate the performance of the model using R-squared
r2 = r2_score(y_test, y_pred)
print('R-squared:', r2)

R-squared: 0.5409147642180705


In [8]:
# Fit the linear regression model on the entire dataset
model = LinearRegression()
model.fit(X, y)

LinearRegression()

In [11]:
import pandas as pd
from sklearn.linear_model import LinearRegression

# Load the dataset from a CSV file
df = pd.read_csv('/Users/sofiatajide/Downloads/Bike-Sharing-Dataset/day.csv')

# Define the independent variables with column names
X = pd.DataFrame(data=df, columns=['season', 'temp', 'hum', 'windspeed', 'weathersit'])

# Define the dependent variable
y = df['cnt']

# Fit the linear regression model on the entire dataset
model = LinearRegression()
model.fit(X, y)

# Predict bike rental demand for a future date
future_date = pd.DataFrame([[2, 0.5, 0.6, 0.3, 1]], columns=['season', 'temp', 'hum', 'windspeed', 'weathersit'])
future_demand = model.predict(future_date)
print('Predicted bike rental demand:', future_demand[0])


Predicted bike rental demand: 4207.193420787011


The predicted bike rental demand of 4207.193420787011 is specific to the input features provided in  future_date DataFrame. These values correspond to a season of 2 (spring), a temperature of 0.5 (normalized), a humidity of 0.6 (normalized), a windspeed of 0.3 (normalized), and a weathersit of 1 (clear or partly cloudy).

It's important to keep in mind that this is just a single prediction based on a specific set of input features. The actual bike rental demand for this future date may be different due to various external factors that could influence demand. However, this prediction can still provide valuable insights and help inform business decisions.